diff --git a/Cargo.toml b/Cargo.toml
index 78eda688..066604cb 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -139 +139,7 @@ prometheus-client = "0.24.0"
-quinn = "0.11.5"
+quinn = { version = "0.11.5", default-features = false, features = [
+  "bloom",
+  "log",
+  "platform-verifier",
+  "runtime-tokio",
+  "rustls-aws-lc-rs",
+] }
@@ -149,0 +156 @@ rustls = { version = "0.23.31", default-features = false, features = [
+  "aws_lc_rs",
@@ -151 +158 @@ rustls = { version = "0.23.31", default-features = false, features = [
-  "ring",
+  "prefer-post-quantum",
@@ -191 +198 @@ thread-priority = "1.2.0"
-tls-listener = { version = "0.11.0", features = ["rustls-ring"] }
+tls-listener = { version = "0.11.0", features = ["rustls-aws-lc"] }
diff --git a/io/zenoh-link-commons/src/quic.rs b/io/zenoh-link-commons/src/quic.rs
index 14816b62..39556680 100644
--- a/io/zenoh-link-commons/src/quic.rs
+++ b/io/zenoh-link-commons/src/quic.rs
@@ -214,2 +214,2 @@ impl<'a> TlsServerConfig<'a> {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -220 +220 @@ impl<'a> TlsServerConfig<'a> {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
@@ -311,2 +311,2 @@ impl<'a> TlsClientConfig<'a> {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -317 +317 @@ impl<'a> TlsClientConfig<'a> {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
diff --git a/io/zenoh-link-commons/src/tls.rs b/io/zenoh-link-commons/src/tls.rs
index b01abdb0..6af28abd 100644
--- a/io/zenoh-link-commons/src/tls.rs
+++ b/io/zenoh-link-commons/src/tls.rs
@@ -83 +83 @@ impl ServerCertVerifier for WebPkiVerifierAnyServerName {
-            &rustls::crypto::ring::default_provider().signature_verification_algorithms,
+            &rustls::crypto::aws_lc_rs::default_provider().signature_verification_algorithms,
@@ -97 +97 @@ impl ServerCertVerifier for WebPkiVerifierAnyServerName {
-            &rustls::crypto::ring::default_provider().signature_verification_algorithms,
+            &rustls::crypto::aws_lc_rs::default_provider().signature_verification_algorithms,
@@ -102 +102 @@ impl ServerCertVerifier for WebPkiVerifierAnyServerName {
-        rustls::crypto::ring::default_provider()
+        rustls::crypto::aws_lc_rs::default_provider()
diff --git a/io/zenoh-links/zenoh-link-quic/src/unicast.rs b/io/zenoh-links/zenoh-link-quic/src/unicast.rs
index 99097d73..b642ed75 100644
--- a/io/zenoh-links/zenoh-link-quic/src/unicast.rs
+++ b/io/zenoh-links/zenoh-link-quic/src/unicast.rs
@@ -382,2 +382,2 @@ impl LinkManagerUnicastTrait for LinkManagerUnicastQuic {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -388 +388 @@ impl LinkManagerUnicastTrait for LinkManagerUnicastQuic {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
diff --git a/io/zenoh-links/zenoh-link-quic/src/utils.rs b/io/zenoh-links/zenoh-link-quic/src/utils.rs
index 0fb589a6..f6ea3f24 100644
--- a/io/zenoh-links/zenoh-link-quic/src/utils.rs
+++ b/io/zenoh-links/zenoh-link-quic/src/utils.rs
@@ -208,2 +208,2 @@ impl<'a> TlsServerConfig<'a> {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -214 +214 @@ impl<'a> TlsServerConfig<'a> {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
@@ -305,2 +305,2 @@ impl<'a> TlsClientConfig<'a> {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -311 +311 @@ impl<'a> TlsClientConfig<'a> {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
diff --git a/io/zenoh-links/zenoh-link-quic_datagram/src/unicast.rs b/io/zenoh-links/zenoh-link-quic_datagram/src/unicast.rs
index 77043ad3..0d01f6db 100644
--- a/io/zenoh-links/zenoh-link-quic_datagram/src/unicast.rs
+++ b/io/zenoh-links/zenoh-link-quic_datagram/src/unicast.rs
@@ -359,2 +359,2 @@ impl LinkManagerUnicastTrait for LinkManagerUnicastQuicDatagram {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -365 +365 @@ impl LinkManagerUnicastTrait for LinkManagerUnicastQuicDatagram {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
diff --git a/io/zenoh-links/zenoh-link-tls/src/utils.rs b/io/zenoh-links/zenoh-link-tls/src/utils.rs
index e09d456b..1b99b1ea 100644
--- a/io/zenoh-links/zenoh-link-tls/src/utils.rs
+++ b/io/zenoh-links/zenoh-link-tls/src/utils.rs
@@ -228,2 +228,2 @@ impl<'a> TlsServerConfig<'a> {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -234 +234 @@ impl<'a> TlsServerConfig<'a> {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
@@ -360,2 +360,2 @@ impl<'a> TlsClientConfig<'a> {
-        // Install ring based rustls CryptoProvider.
-        rustls::crypto::ring::default_provider()
+        // Install AWS-LC based rustls CryptoProvider.
+        rustls::crypto::aws_lc_rs::default_provider()
@@ -366 +366 @@ impl<'a> TlsClientConfig<'a> {
-            // Ignore the error here, because `rustls::crypto::ring::default_provider().install_default()` will inevitably be executed multiple times
+            // Ignore the error here, because provider installation will inevitably be executed multiple times
